### PR TITLE
cmd/charm/charmcmd: add output to push about channel and force revision on publish.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/41/
Test PASSed.

**Comment by fabricematrat** [comment]
QA:
https://github.com/juju/charmstore-client/issues/16
https://github.com/juju/charmstore-client/issues/18

Now charm publish force the revision so no need to specify any ouput.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/43/
Test FAILed.

**Comment by urosj** [comment]
QA :+1: 
Once tests pass, it is ready to land.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/46/
Test PASSed.

**Comment by fabricematrat** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-client-merge

File: cmd/charm/charmcmd/publish.go [hunk 1]
```diff
@@ -33,9 +33,10 @@ type publishCommand struct {
 var publishDoc = `
 The publish command publishes a charm or bundle in the charm store.
 Publishing is the action of assigning one channel to a specific charm
-or bundle revision, so that it can be shared with other users and also
-referenced without specifying the revision. Two channels are supported:
-"stable" and "development", the "stable" channel is used by default.
+or bundle revision (revision need to be specified), so that it can be shared
+with other users and also referenced without specifying the revision.
+Two channels are supported: "stable" and "development", the "stable" channel is
```
Comment (rogpeppe) [inline cmd/charm/charmcmd/publish.go]: s/,/;/

File: cmd/charm/charmcmd/publish.go [hunk 2]
```diff
@@ -79,6 +80,10 @@ func (c *publishCommand) Init(args []string) error {
 	if err != nil {
 		return errgo.Notef(err, "invalid charm or bundle id")
 	}
+	if id.Revision == -1 {
+		return errgo.Newf("revision need to be specified")
```
Comment (rogpeppe) [inline cmd/charm/charmcmd/publish.go]: s/need/needs/

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/charmstore-client-merge/14

File: cmd/charm/charmcmd/publish.go [hunk 3]
```diff
@@ -105,6 +110,8 @@ func (c *publishCommand) Run(ctxt *cmd.Context) error {
 	if err != nil {
 		return errgo.Notef(err, "cannot publish charm or bundle")
 	}
+	fmt.Fprintln(ctxt.Stdout, "url:", c.id)
```
Comment (rogpeppe) [inline cmd/charm/charmcmd/publish.go:117]: If the revision needs to be specified, this really is only printing what we specified on the command line, so meh.

If we really need to print something, perhaps:

fmt.Fprintf(ctxt.Stdout, "%s published to channel %s\n", c.id, c.channel)

?

**Comment by rogpeppe** [comment]
LGTM with a couple of minor suggestions, thanks.

File: cmd/charm/charmcmd/publish.go (hunk 3 above)
Comment (fabricematrat) [inline cmd/charm/charmcmd/publish.go:117]: this is to replicate the push command output

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/48/
Test PASSed.

**Comment by fabricematrat** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-client-merge

**Pull request merged**